Modules/CalcOffence.lua

```diff
@@ -2134,6 +2134,18 @@ function calcs.offence(env, actor, activeSkill)
 	output.ManaLeechInstanceRate = output.Mana * data.misc.LeechRateBase * calcLib.mod(skillModList, skillCfg, "ManaLeechRate")
 	output.ManaLeechRate = output.ManaLeechInstantRate + m_min(output.ManaLeechInstances * output.ManaLeechInstanceRate, output.MaxManaLeechRate) * output.ManaRecoveryRateMod
 	output.ManaLeechPerHit = output.ManaLeechInstant + m_min(output.ManaLeechInstanceRate, output.MaxManaLeechRate) * output.ManaLeechDuration * output.ManaRecoveryRateMod
+	
+	if skillModList:Flag(nil, "ImmortalAmbition") then
+		local val1 = output.EnergyShieldLeechInstanceRate 
+		local val2 = output.EnergyShieldLeechRate 
+		local val3 = output.EnergyShieldLeechPerHit 
```

Comment:
small nit. val 1 unused and not very descriptive or needed names

Reply:
Got it, thx